lib/data/wrapper.js

```diff
@@ -61,61 +67,81 @@ const data = {
             writeStream = cipherBundle.cipher;
             hashedStream.pipe(writeStream);
         }
-
-        client.put(writeStream, valueSize, keyContext, log.getSerializedUids(),
-           (err, key) => {
-               if (err) {
-                   log.error('error from datastore',
+        if (implName === 'multipleBackends') {
+            // Need to send backendInfo to client.put and
+            // client.put will provide dataRetrievalInfo so no
+            // need to construct here
+            return client.put(writeStream, valueSize, keyContext, backendInfo,
+                log.getSerializedUids(), (err, dataRetrievalInfo) => {
+                    if (err) {
+                        log.error('error from datastore',
+                                 { error: err, implName });
+                        return cb(errors.InternalError);
+                    }
+                    return cb(null, dataRetrievalInfo, hashedStream);
+                });
+        }
+        return client.put(writeStream, valueSize, keyContext,
+            log.getSerializedUids(), (err, key) => {
+                if (err) {
+                    log.error('error from datastore',
                              { error: err, implName });
-                   return cb(errors.InternalError);
-               }
-               const dataRetrievalInfo = {
-                   key,
-                   dataStoreName: implName,
-               };
-               return cb(null, dataRetrievalInfo, hashedStream);
-           });
+                    return cb(errors.InternalError);
+                }
+                const dataRetrievalInfo = {
+                    key,
+                    dataStoreName: implName,
+                };
+                return cb(null, dataRetrievalInfo, hashedStream);
+            });
     },
 
     get: (objectGetInfo, log, cb) => {
         // If objectGetInfo.key exists the md-model-version is 2 or greater.
         // Otherwise, the objectGetInfo is just the key string.
         const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
+        // const implNameGetInfo = (implName === 'sproxyd') ?
```

Comment:
I've uncommented both of these lines. They're needed because sproxyd expects a key string instead of an object